lms/djangoapps/shoppingcart/views.py

```diff
@@ -352,8 +351,15 @@ def register_code_redemption(request, registration_code):
         course = get_course_by_id(getattr(course_registration, 'course_id'), depth=0)
         if reg_code_is_valid and not reg_code_already_redeemed:
             #now redeem the reg code.
-            RegistrationCodeRedemption.create_invoice_generated_registration_redemption(course_registration, request.user)
-            CourseEnrollment.enroll(request.user, course.id)
+            redemption = RegistrationCodeRedemption.create_invoice_generated_registration_redemption(course_registration, request.user)
+
+            kwargs = {}
+            if course_registration.mode_slug is not None:
+                kwargs['mode'] = course_registration.mode_slug
+
+            redemption.course_enrollment = CourseEnrollment.enroll(request.user, course.id, **kwargs)
```

Comment:
I know this isn't part of your PR, but I'm wondering if we're being careful enough with this call.  In particular:

1) Are we handling all the exceptions this might raise?
2) Should we be passing `check_access=True` to prevent users from enrolling outside of the enrollment period?  Or are we explicitly ignoring these constraints for redeem codes?  (that may be a question for product)